### ISSUE TITLE
Bug for PGnet Inference

### ISSUE THREAD

**Issue by Daisy5296** [PaddlePaddle/PaddleOCR#10158]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：CUDA11.7 Pytorch1.13
- 版本号/Version：Paddle：2.4.2.post117  PaddleOCR：2.6.1.3 问题相关组件/Related components：PGNet & train & inference
- 运行指令/Command Code：python3 -m paddle.distributed.launch --gpus '0,1' tools/train.py -c configs/e2e/e2e_r50_vd_pg.yml
- 完整报错/Complete Error Message：
Hi, when I train PGnet, I found that the metric of evaluation during training almost does not increase when the training loss decrease a lot, even when I am using the same dataset for training and evaluation. 

Also, I visualize some of the intermediate steps. For a specific picture, the network inference output during training, for example, tcl, is visually correct with score_loss of 0.1. However, when I use the same picture and the same model for evaluation, the tcl output is totally wrong visually. The same case happens for the inference code.

It looks like there are bugs for the evaluation during training and also the inference code. 

Could you please help to have a look?

**Comment by andyjpaddle** [comment]
pls try the latest code in `dygraph` branch

**Comment by XinDing5** [comment]
Hi thanks for reply! I've tried the dygraph branch, and it makes no difference on this issue. 

Also, I make some more experiments. I use a 5000 pics dataset as both the training set and validation set. I keep only the score_loss and set the other losses to zero（i.e., only train the tcl branch）. The training converges well after 2 epochs:

epoch: [1/600], global_step: 10, lr: 0.001000, loss: 0.964902, score_loss: 0.964902, border_loss: 0.000000, direction_loss: 0.000000, ctc_loss: 0.000000, avg_reader_cost: 0.61512 s, avg_batch_cost: 1.27937 s, avg_samples: 14.0, ips: 10.94291 samples/s, eta: 3 days, 4:07:07
…
[2023/06/14 16:33:48] ppocr INFO: epoch: [1/600], global_step: 357, lr: 0.001000, loss: 0.077057, score_loss: 0.077057, border_loss: 0.000000, direction_loss: 0.000000, ctc_loss: 0.000000, avg_reader_cost: 0.65975 s, avg_batch_cost: 0.84894 s, avg_samples: 9.8, ips: 11.54386 samples/s, eta: 3 days, 0:48:59
…
[2023/06/14 16:43:48] ppocr INFO: epoch: [2/600], global_step: 714, lr: 0.001000, loss: 0.080125, score_loss: 0.080125, border_loss: 0.000000, direction_loss: 0.000000, ctc_loss: 0.000000, avg_reader_cost: 0.44298 s, avg_batch_cost: 0.55222 s, avg_samples: 5.6, ips: 10.14096 samples/s, eta: 3 days, 0:30:49

The loss seems to be small enought. However, the evaluation results are still not good. The results after the first and second epochs are:

ppocr INFO: cur metric, total_num_gt: 5000, total_num_det: 294, global_accumulative_recall: 0, hit_str_count: 0, recall: 0.0, precision: 0.0, f_score: 0, seqerr: 1, recall_e2e: 0.0, precision_e2e: 0.0, f_score_e2e: 0, fps: 38.65965607935255

ppocr INFO: cur metric, total_num_gt: 5000, total_num_det: 0, global_accumulative_recall: 0, hit_str_count: 0, recall: 0.0, precision: 0, f_score: 0, seqerr: 1, recall_e2e: 0.0, precision_e2e: 0, f_score_e2e: 0, fps: 39.267057566202325

I visualize the tcl label and the model's tcl output during the training process, and they look correct:
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/29244196/cad90ae0-30c3-443d-bb16-3614bc5d2a48)
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/29244196/da299c6f-d19d-4a3e-b4e4-004d2e50868a)

But the model's tcl output during the evaluation process is incorrect, detects nothing:
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/29244196/45458ca8-7f85-43b5-89e9-939aeccce638)

Looks like bugs exist in the evaluation code. Could you please help to check? 

Many thanks!

**Comment by andyjpaddle** [comment]
try change this [param](https://github.com/PaddlePaddle/PaddleOCR/blob/dygraph/configs/e2e/e2e_r50_vd_pg.yml#L108) to 512

**Comment by XinDing5** [comment]
Hi I've tried changing the param to 512, and the results are similar to before, except that the tcl output during evaluation becomes 96*128, which was 128*192 when the param was 768.
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/29244196/efe4168d-e28e-4c47-a740-bcd8c50f1419)

FYI, I save this tcl output from here (p_score[0]): https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/utils/e2e_utils/pgnet_pp_utils.py#L60

Is it that the data pre-process incorrect somehow?

**Comment by XinDing5** [comment]
For anyone who is interested, I can now confirm that there is a bug in the PGNet inference code. During training, the input pictures are actually in RGB mode, even though it is stated as BGR in the config. This can be found as the implementation in: https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/data/imaug/pg_process.py#L1016

But in the evaluation during training and the inference process, the input pictures are in BGR mode, as implemented in: https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/data/imaug/operators.py

This bug exists in both dygraph and 2.6 versions.

**Comment by Daisy5296** [comment]
Again, for anyone who is interested, there is also a bug in the e2e metric calculation code. It forces the ground truth text to lower case, which is incorrect in my case. As implemented in:
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/data/imaug/label_ops.py#L226
